pkg/v1/manifest.go

```diff
@@ -37,6 +37,15 @@ type Descriptor struct {
 	Digest      Hash              `json:"digest"`
 	URLs        []string          `json:"urls,omitempty"`
 	Annotations map[string]string `json:"annotations,omitempty"`
+	Platform    *Platform         `json:"platform,omitempty"`
+}
+
+// Index represents an OCI image index in a structured way.
+type Index struct {
+	SchemaVersion int64             `json:"schemaVersion"`
+	MediaType     types.MediaType   `json:"mediaType"`
+	Manifests     []Descriptor      `json:"manifests"`
+	Annotations   map[string]string `json:"annotations,omitempty"`
```

Comment:
nit: we have `config.go` and `manifest.go`...

Reply:
Moved some things around. I think it kind of makes sense to keep the index in `manifest.go`, since it's a _kind_ of manifest, but it's not a strong opinion.